test/unit/math/prim/mat/prob/normal_test.cpp

```diff
@@ -0,0 +1,146 @@
+#include <stan/math/prim/scal/meta/max_size.hpp>
+#include <gtest/gtest.h>
+#include <boost/math/distributions.hpp>
+#include <boost/random/mersenne_twister.hpp>
+#include <test/unit/math/prim/scal/prob/util.hpp>
+
+using Eigen::Dynamic;
+using Eigen::Matrix;
+
+TEST(ProbDistributionsNormal, error_check_vec) {
+  boost::random::mt19937 rng;
+
+  double mu = 1.0;
+  double sigma = 2.0;
+  Matrix<double, Dynamic, 1> mu_vec(3), sigma_vec(3),
+    mu_vec2(4), sigma_vec2(4);
+
+  mu_vec << -1.0, 2.0, 3.0;
+  sigma_vec << 1.0, 2.0, 3.0;
+  mu_vec2 << 1.0, -2.0, 3.0, 4.0;
+  sigma_vec2 << 1.0, 2.0, 3.0, 4.0;
+
+  EXPECT_NO_THROW(stan::math::normal_rng(mu, sigma_vec, rng));
+  EXPECT_NO_THROW(stan::math::normal_rng(mu_vec, sigma, rng));
+  EXPECT_NO_THROW(stan::math::normal_rng(mu_vec, sigma_vec, rng));
+
+  mu = stan::math::positive_infinity();
+  mu_vec[1] = stan::math::positive_infinity();
+
+  EXPECT_THROW(stan::math::normal_rng(mu, sigma_vec, rng),
+               std::domain_error);
+  EXPECT_THROW(stan::math::normal_rng(mu_vec, sigma, rng),
+               std::domain_error);
+  EXPECT_THROW(stan::math::normal_rng(mu_vec, sigma_vec, rng),
+               std::domain_error);
+
+  mu = stan::math::negative_infinity();
+  mu_vec[1] = 1.0;
+  mu_vec[0] = stan::math::negative_infinity();
+  EXPECT_THROW(stan::math::normal_rng(mu, sigma_vec, rng),
+               std::domain_error);
+  EXPECT_THROW(stan::math::normal_rng(mu_vec, sigma, rng),
+               std::domain_error);
+  EXPECT_THROW(stan::math::normal_rng(mu_vec, sigma_vec, rng),
+               std::domain_error);
+
+  mu = 0.0;
+  mu_vec[0] = 0.0;
+  sigma = 0.0;
+  sigma_vec[0] = 0.0;
+  EXPECT_THROW(stan::math::normal_rng(mu, sigma_vec, rng),
+               std::domain_error);
+  EXPECT_THROW(stan::math::normal_rng(mu_vec, sigma, rng),
+               std::domain_error);
+  EXPECT_THROW(stan::math::normal_rng(mu_vec, sigma_vec, rng),
+               std::domain_error);
+
+  sigma = -2.0;
+  sigma_vec[0] = 1.0;
+  sigma_vec[1] = -2.0;
+  EXPECT_THROW(stan::math::normal_rng(mu, sigma_vec, rng),
+               std::domain_error);
+  EXPECT_THROW(stan::math::normal_rng(mu_vec, sigma, rng),
+               std::domain_error);
+  EXPECT_THROW(stan::math::normal_rng(mu_vec, sigma_vec, rng),
+               std::domain_error);
+
+  sigma = 1.0;
+  sigma_vec[1] = 1.0;
+  EXPECT_THROW(stan::math::normal_rng(mu_vec, sigma_vec2, rng),
+               std::invalid_argument);
+  EXPECT_THROW(stan::math::normal_rng(mu_vec2, sigma_vec, rng),
+  std::invalid_argument);
+}
+
+template<typename T1, typename T2>
+void chiSquareTestNormal(const T1& mu,
+                         const T2& sigma) {
+  boost::random::mt19937 rng;
+  int N = 10000;
+  int M = stan::max_size(mu, sigma);
+  std::vector<std::vector<double> > samples_to_test_transpose;
+
+  for (int n = 0; n < N; ++n) {
+    samples_to_test_transpose.push_back(stan::math::normal_rng(mu, sigma, rng));
```

Comment:
This looks like the output is `std::vector<double>`. 

I'm thinking it would be more useful as `Eigen::VectorXd` because that tends to be usable in more contexts.

If you think it's better as `vector<double>` I'm OK with that.  I don't think there should be multiple versions given that we have functions to convert.

I do think `normal_rng` should take any kind of container argument, but I didn't go and look at that part of the pull request.